sqs-active-job/lib/sqs_active_job/poller.rb

```diff
@@ -22,17 +22,11 @@ class Poller
 
         def initialize(args = ARGV)
           @options = parse_args(args)
-          # Set_environment must be run before we boot_rails
-          set_environment
-        end
-
-        def set_environment
-          @environment = @options[:environment] || ENV['APP_ENV'] || ENV['RAILS_ENV'] || ENV['RACK_ENV'] || 'development'
+          @options[:queue] ||= :default
+          # please set environment and boot rails BEFORE initializing this poller
         end
 
         def run
-          # exit 0
-          boot_rails
```

Comment:
I'm not sure the best way to do this, but what about trying to require rails and running boot_rails only when its available? something like:
```
begin
  require 'rails'
  boot_rails
rescue LoadError
 # no boot
end
```

Reply:
Thanks for reviewing.

I moved the Rails booting to the executable which I think is a better place for it. The poller doesn't need to control what environment it's polling in. If you really want the poller to do booting we could split the poll method to boot and then call poll_without_booting_rails. This new method poll_without_booting_rails could then be used by non-rails environments that want to do polling. Other environments can easily create their own executables if desired.